analyzers/tests/SonarAnalyzer.Test/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -47,6 +46,313 @@ public void InstanceReference_SetsNotNull_VB()
+                    Tag("After2", collection);
+                }
+
+                private IEnumerable<int> Untracked() => [ 1, 2, 3 ];
+
+                private static void Tag(string name, object arg) { }
+            }
+            """;
+        var validator = new SETestContext(code, AnalyzerLanguage.CSharp, Array.Empty<SymbolicCheck>()).Validator;
+        validator.ValidateContainsOperation(OperationKind.Invocation);
+        validator.TagValue("Before1").Should().BeNull();
+        validator.TagValue("After1").Should().HaveOnlyConstraint(ObjectConstraint.NotNull);
+        validator.TagValue("Before2").Should().BeNull();
+        validator.TagValue("After2").Should().HaveOnlyConstraint(ObjectConstraint.NotNull);
+    }
+
+    [DataTestMethod]
+    [DataRow("collection.Cast<short>()")]
+    [DataRow("collection.OfType<short>()")]
+    [DataRow("collection.ThenBy(x => x)")]
+    [DataRow("collection.ThenByDescending(x => x)")]
+    public void Invocation_SetsNotNull_OnLinqMethodsWithGenericParamNullChecking(string invocation)
+    {
+        var code = $$"""
+            using System;
+            using System.Linq;
+            using System.Collections.Generic;
+            public class Sample
+            {
+                public void Main(IOrderedEnumerable<int> collection)
+                {
+                    Tag("Before1", collection);
+                    _ = {{invocation}};
+                    Tag("After1", collection);
+                }
+
+                private static void Tag(string name, object arg) { }
+            }
+            """;
+        var validator = new SETestContext(code, AnalyzerLanguage.CSharp, Array.Empty<SymbolicCheck>()).Validator;
+        validator.ValidateContainsOperation(OperationKind.Invocation);
+        validator.TagValue("Before1").Should().BeNull();
+        validator.TagValue("After1").Should().HaveOnlyConstraint(ObjectConstraint.NotNull);
+    }
+
+    [DataTestMethod]
+    [DataRow("OrderBy", "x => x")]
+    [DataRow("OrderByDescending", "x => x")]
+    [DataRow("GroupBy", "x => x")]
+    [DataRow("AsEnumerable")]
+    [DataRow("ToLookup", "x => x, x => x")]
+    public void Invocation_DoesNotSetNotNull_OnLinqMethodNonNullChecking(string method, string args = null)
+    {
+        var code = $$"""
+            using System;
+            using System.Linq;
+            using System.Collections.Generic;
+            public class Sample
+            {
+                public void Main(IEnumerable<int> collection)
+                {
+                    Tag("Before1", collection);
+                    _ = collection.{{method}}({{args}});                                                 // Call as extension
+                    Tag("After1", collection);
+
+                    collection = Untracked();
+                    Tag("Before2", collection);
+                    _ = Enumerable.{{method}}(collection{{(args is null ? string.Empty : "," + args)}}); // Call as normal static method
```

Comment:
The ternary is overkill. Just move the `,` into the datarows, like `", x => x"`

Reply:
Then it's not going to work 5 lines above, in `_ = collection.{{method}}({{args}});`.
Removing the `,` there, rather than adding here, is worse.